salt/_modules/kubernetes.py

```diff
@@ -1975,3 +1975,178 @@ def replace_daemonset(
             raise CommandExecutionError(exc)
     finally:
         _cleanup(**cfg)
+
+
+def show_clusterrole(name, **kwargs):
+    cfg = _setup_conn(**kwargs)
+    try:
+        api_instance = kubernetes.client.RbacAuthorizationV1Api()
+        api_response = api_instance.read_cluster_role(name)
+
+        return api_response.to_dict()
+    except (ApiException, HTTPError) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        else:
+            log.exception(
+                'Exception when calling '
+                'RbacAuthorizationV1Api->read_cluster_role'
+            )
+            raise CommandExecutionError(exc)
+    finally:
+        _cleanup(**cfg)
+
+
+def create_clusterrole(
+        name,
+        rules,
+        **kwargs):
+    meta_obj = kubernetes.client.V1ObjectMeta(name=name)
+    body = kubernetes.client.V1ClusterRole(
+        metadata=meta_obj,
+        rules=rules)
+
+    cfg = _setup_conn(**kwargs)
+
+    try:
+        api_instance = kubernetes.client.RbacAuthorizationV1Api()
+        api_response = api_instance.create_cluster_role(
+            body=body)
+
+        return api_response.to_dict()
+    except (ApiException, HTTPError) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        else:
+            log.exception(
+                'Exception when calling '
+                'RbacAuthorizationV1Api->create_cluster_role'
+            )
+            raise CommandExecutionError(exc)
+    finally:
+        _cleanup(**cfg)
+
+
+def replace_clusterrole(
+        name,
+        rules,
+        **kwargs):
+    meta_obj = kubernetes.client.V1ObjectMeta(name=name)
+    body = kubernetes.client.V1ClusterRole(
+        metadata=meta_obj,
+        rules=rules)
+
+    cfg = _setup_conn(**kwargs)
+
+    try:
+        api_instance = kubernetes.client.RbacAuthorizationV1Api()
+        api_response = api_instance.replace_cluster_role(name, body)
+
+        return api_response.to_dict()
+    except (ApiException, HTTPError) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        else:
+            log.exception(
+                'Exception when calling '
+                'RbacAuthorizationV1Api->replace_cluster_role'
+            )
+            raise CommandExecutionError(exc)
+    finally:
+        _cleanup(**cfg)
+
+
+def show_customresourcedefinition(name, **kwargs):
+    cfg = _setup_conn(**kwargs)
```

Comment:
I'm surprised this (and `replace_crd`) could succeed without the monkey-patch being in place. Anyway, see https://github.com/scality/metalk8s/pull/696/files#r261885992, must be placed elsewhere anyway.

Reply:
`show` don't need the workaround for replace I got another issue will look at it but it's not the same